dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -2416,6 +2416,41 @@ global using nfloat = global::System.Runtime.InteropServices.NFloat%3B
 		</ItemGroup>
 	</Target>
 
+	<Target Name="ComputeXcsyncArguments">
+		<PropertyGroup>
+			<xcSyncExecute>$(DOTNET_HOST_PATH) $(XcsyncPath)</xcSyncExecute>
+			<xcSyncProjectFile Condition="'$(xcSyncProjectFile)' == ''">$(MSBuildProjectDirectory)/$(MSBuildProjectFile)</xcSyncProjectFile>
+			<xcSyncXcodeFolder Condition="'$(xcSyncXcodeFolder)' == ''">$(IntermediateOutputPath)xcode</xcSyncXcodeFolder>
+			<xcSyncTargetFrameworkMoniker Condition="'$(xcSyncTargetFrameworkMoniker)' == ''">$(TargetFramework)</xcSyncTargetFrameworkMoniker>
+			<xcSyncVerbosity Condition="'$(xcSyncVerbosity)' == ''">Diagnostic</xcSyncVerbosity>
+
+
+			<XcsyncCommonArguments Condition="'$(xcSyncProjectFile)' != ''">$(XcsyncCommonArguments) -p $(xcSyncProjectFile)</XcsyncCommonArguments>
+			<XcsyncCommonArguments Condition="'$(xcSyncXcodeFolder)' != ''">$(XcsyncCommonArguments) -t $(xcSyncXcodeFolder)</XcsyncCommonArguments>
+			<XcsyncCommonArguments Condition="'$(xcSyncTargetFrameworkMoniker)' != ''">$(XcsyncCommonArguments) -tfm $(xcSyncTargetFrameworkMoniker)</XcsyncCommonArguments>
+			<XcsyncCommonArguments Condition="'$(xcSyncVerbosity)' != ''">$(XcsyncCommonArguments) -v $(xcSyncVerbosity)</XcsyncCommonArguments>
+			<XcsyncCommonArguments>$(XcsyncCommonArguments) --dotnet-path $([System.IO.Path]::GetDirectoryName($(DOTNET_HOST_PATH)))</XcsyncCommonArguments>
+
+			<XcsyncGenerateArguments>$(XcsyncCommonArguments) --force</XcsyncGenerateArguments>
+
+			<XcsyncSyncArguments>$(XcsyncCommonArguments)</XcsyncSyncArguments>
+		</PropertyGroup>
+	</Target>
+
+	<Target Name="xcsync-generate" DependsOnTargets="ComputeXcsyncArguments">
+		<Message Importance="Normal" Text="Add $(XcsyncPath) to your path to invoke xcSync directly from the commandline." />
```

Comment:
Do we intend for people to run xcsync themselves? If so, it might make more sense to ship as a tool pack within the workload: https://github.com/dotnet/designs/blob/main/accepted/2020/workloads/workload-manifest.md#tools-packs

Also the message is wrong, because `XcsyncPath` is the path to the xcsync.dll, and you'd only add the parent folder to PATH, not the dll itself.

Reply:
tool-packs aren't implemented yet according to @jonathanpeppers

Reply:
`tools` are in the workload spec, but they didn't implement it. The part that is missing is where it would automatically setup the tool to be invoked like `dotnet-foo`, `dotnet foo`, `foo`, etc. Today, you can make an SDK pack with tools inside, like the AOT compilers do.